main.nf

```diff
@@ -203,36 +203,71 @@ process bwa {
     module 'bwa'
     module 'samtools'
     module 'BEDTools'
-    
+
     cpus 2
     memory { 16.GB * task.attempt }
     time { 120.h * task.attempt }
     errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
     maxRetries 3
     maxErrors '-1'
-    
+
     publishDir "${params.outdir}/bwa", mode: 'copy'
-    
+
     input:
     file (reads:'*') from trimmed_reads
     set val(prefix) from name_for_bwa
 
     output:
+    file '*.sam' into bwa_sam
+    stdout into bwa_logs
+
+    script:
+    """
+    bwa mem -M $index $reads > ${prefix}.sam
+    """
+}
+
+/*
+ * STEP 3.2 - post-alignment processing
+ */
+
+process samtools {
+    tag "$prefix"
+
+    module 'bioinfo-tools'
+    module 'samtools'
+    module 'BEDTools'
+
+    cpus 2
+    memory { 16.GB * task.attempt }
+    time { 120.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
+
+    publishDir "${params.outdir}/bwa", mode: 'copy'
+
+    input:
+    file bwa_sam
+    set val(prefix) from name_for_samtools
+
+    output:
     file '*.sorted.bam' into bam_picard
     file '*.sorted.bam.bai'
-    file '*.sorted.bed'
+    file '*.sorted.bed' into bed_total
     stdout into bwa_logs
```

Comment:
Getting closer! This needs its own channel name now though, it's now a duplicate. So `samtools_logs`, then an extra line for MultiQC below.